src/pages/ticketConfirmation.tsx

```diff
@@ -0,0 +1,337 @@
+import { ServiceListAttribute } from './api/serviceList';
+import { NumberOfProductsAttribute } from './api/howManyProducts';
+import { MultipleProductAttribute } from './api/multipleProductValidity';
+
+const title = 'Ticket Confirmation - Fares Data Build Tool';
+const description = 'Ticket Confirmation page of the Fares Data Build Tool';
+
+export type TicketConfirmationProps = {
+    fareTypeProps: SingleTicketProps | ReturnTicketProps | PeriodTicketProps | FlatFareTicketProps;
+};
+
+export interface SingleTicketProps {
+    service: string;
+    journeyDirection: string;
+    matchedFareStages: MatchedFareStages[];
+}
+
+export interface MatchedFareStages {
+    fareStage: string;
+    stops: string[];
+}
+
+export interface ReturnTicketProps {
+    service: string;
+    circular: boolean;
+    inboundMatchedFareStages: MatchedFareStages[];
+    outboundMatchedFareStages: MatchedFareStages[];
+    validity?: ReturnPeriodValidity;
+}
+export interface PeriodTicketProps {
+    services: string[];
+    zone: boolean;
+    numberOfProducts: number;
+    products: Product[];
+}
+export interface FlatFareTicketProps {
+    services: string[];
+    productName: string;
+    productPrice: string;
+}
+
+export const buildMatchedFareStages = (matchingFareZones: MatchingFareZones): MatchedFareStages[] => {
+    const matchedFareStages: MatchedFareStages[] = [];
+    const entries = Object.entries(matchingFareZones);
+    entries.forEach(entry => {
+        if (entry[0] !== 'undefined') {
+            const stopList = entry[1].stops;
+            const stops: string[] = stopList.map(stop => stop.stopName);
+            matchedFareStages.push({ fareStage: entry[1].name, stops });
+        }
+    });
+    return matchedFareStages;
+};
+
+export const buildFareTypeProps = (
+    fareType: string,
+    ctx: NextPageContextWithSession,
+): SingleTicketProps | ReturnTicketProps | PeriodTicketProps | FlatFareTicketProps => {
+    if (fareType === 'single') {
+        const { service } = getSessionAttribute(ctx.req, SERVICE_ATTRIBUTE) as Service;
+        const journeyDirection = (getSessionAttribute(ctx.req, JOURNEY_ATTRIBUTE) as Journey).directionJourneyPattern;
+        if (!journeyDirection) {
+            throw new Error('User has no journey direction information.');
+        }
+        const matchedFareStages: MatchedFareStages[] = buildMatchedFareStages(
+            (getSessionAttribute(ctx.req, MATCHING_ATTRIBUTE) as MatchingInfo).matchingFareZones,
+        );
+        return {
+            service,
+            journeyDirection,
+            matchedFareStages,
+        };
+    }
+    if (fareType === 'return') {
+        const { service } = getSessionAttribute(ctx.req, SERVICE_ATTRIBUTE) as Service;
+        const { inboundJourney } = getSessionAttribute(ctx.req, JOURNEY_ATTRIBUTE) as Journey;
+        const outBoundJourney = (getSessionAttribute(ctx.req, JOURNEY_ATTRIBUTE) as Journey).outboundJourney;
```

Comment:
could we pull both journeys out at the same time?` const { inboundJourney, outBoundJourney } = getSessionAttribute(ctx.req, JOURNEY_ATTRIBUTE) as Journey;`